binding/webgl_rendering_context.cc

```diff
@@ -2466,6 +2576,82 @@ napi_value WebGLRenderingContext::Uniform2i(napi_env env,
   return nullptr;
 }
 
+/* static */
+napi_value WebGLRenderingContext::Uniform2iv(napi_env env,
+                                             napi_callback_info info) {
+  LOG_CALL("Uniform2iv");
+  napi_status nstatus;
+
+  size_t argc = 2;
+  napi_value args[2];
+  napi_value js_this;
+  nstatus = napi_get_cb_info(env, info, &argc, args, &js_this, nullptr);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+  ENSURE_ARGC_RETVAL(env, argc, 2, nullptr);
+
+  ENSURE_VALUE_IS_NUMBER_RETVAL(env, args[0], nullptr);
+
+  GLint location;
+  nstatus = napi_get_value_int32(env, args[0], &location);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  size_t size;
+  void *data;
+  nstatus = napi_get_typedarray_info(env, args[1], nullptr, &size, &data,
+                                     nullptr, nullptr);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  WebGLRenderingContext *context = nullptr;
+  nstatus = UnwrapContext(env, js_this, &context);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  context->eglContextWrapper_->glUniform2iv(location, 1,
+                                            reinterpret_cast<GLint *>(data));
+
+#if DEBUG
+  context->CheckForErrors();
+#endif
+  return nullptr;
+}
+
+/* static */
+napi_value WebGLRenderingContext::Uniform3iv(napi_env env,
+                                             napi_callback_info info) {
+  LOG_CALL("Uniform3iv");
+  napi_status nstatus;
+
+  size_t argc = 2;
+  napi_value args[2];
+  napi_value js_this;
+  nstatus = napi_get_cb_info(env, info, &argc, args, &js_this, nullptr);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+  ENSURE_ARGC_RETVAL(env, argc, 2, nullptr);
+
+  ENSURE_VALUE_IS_NUMBER_RETVAL(env, args[0], nullptr);
+
+  GLint location;
+  nstatus = napi_get_value_int32(env, args[0], &location);
+  ENSURE_NAPI_OK_RETVAL(env, nstatus, nullptr);
+
+  size_t size;
+  void *data;
+  nstatus = napi_get_typedarray_info(env, args[1], nullptr, &size, &data,
```

Comment:
You'll need to check the type here. According to the specs - value can also be a regular array:

https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/uniform

Reply:
My gut was to include this originally, but I didn't see (naive eye here) any usages.  Can you point me to one so I can emulate?

Reply:
You'll need to check for "is this an object" and "is this a typed array?". You'll also need to check for if this is an array.

Pseudo-code:
```c++

bool is_array = false;
napi_is_array(..., args[1], &is_arary, ...);
if (is_array) {
  // Extract buffer:
} else {
  //  Check for typed array:
  
}

```

Reply:
>`nstatus = napi_get_arraybuffer_info(env, args[1], &data, &size);`

When I try this using a standard array, it fails, as I believe `napi_get_arraybuffer_info` is trying to handle a buffer, but what we have is actually a plain javascript array.

Reply:
Added `TODO` to satisfy comment from https://github.com/google/node-gles/pull/4#issuecomment-458769624

Reply:
Ditto here about type checking -